### PR TITLE
fix(debug-texture): crop uv texture

### PR DESCRIPTION
**Pull request by mspivak-actionengine**

### CODE REVIEW

File: src/utils/debug/texture-selector-utils.ts [hunk 1]
```diff
@@ -42,14 +63,29 @@ export function selectDebugTextureForTile(tile, uvDebugTexture) {
...
     }
-    material.pbrMetallicRoughness.baseColorTexture.texture.source.image =
-      uvDebugTexture;
-    tile.content.material = { ...tile.content.material };
+    const width =
+      material.pbrMetallicRoughness.baseColorTexture.texture.source.image.width;
+    const height =
+      material.pbrMetallicRoughness.baseColorTexture.texture.source.image
+        .height;
+    cropTexture(uvDebugTexture, width, height).then((uvDebugTexture) => {
```
Comment (belom88) [inline src/utils/debug/texture-selector-utils.ts:71]: Make `selectDebugTextureForTile` also async. skipping tests is not a way

File: src/utils/debug/texture-selector-utils.spec.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ describe("Texture Selector Utils - selectDebugTextureForTileset", () => {
     expect(result).toBeUndefined();
   });
 
-  test("Should return undefined if no uvDebugTexture", () => {
+  test.skip("Should return undefined if no uvDebugTexture", () => {
```
Comment (belom88) [inline src/utils/debug/texture-selector-utils.spec.ts:16]: Add a test / checks for cropping

**Review by belom88 — CHANGES_REQUESTED** [review]

File: src/utils/debug/texture-selector-utils.ts (hunk 1 above)
Comment (mspivak-actionengine) [reply src/utils/debug/texture-selector-utils.ts]: Done

File: src/utils/debug/texture-selector-utils.spec.ts (hunk 2 above)
Comment (mspivak-actionengine) [reply src/utils/debug/texture-selector-utils.spec.ts]: Done

File: src/utils/debug/texture-selector-utils.ts [hunk 3]
```diff
@@ -22,7 +22,31 @@ export function selectOriginalTextureForTileset() {
   }
 }
 
-export function selectDebugTextureForTile(tile, uvDebugTexture) {
+const cropTexture = async (texture, width, height) => {
```
Comment (belom88) [inline src/utils/debug/texture-selector-utils.ts:25]: Add TSDoc comments and types for the function

File: src/utils/debug/texture-selector-utils.ts [hunk 4]
```diff
@@ -4,15 +4,15 @@
 // We need to store tiles when we replace texture
 const tiles = {};
 
-export function selectDebugTextureForTileset(tileset, uvDebugTexture) {
+export async function selectDebugTextureForTileset(tileset, uvDebugTexture) {
```
Comment (belom88) [inline src/utils/debug/texture-selector-utils.ts:7]: Could you add TSDoc comments and types for the function? It was ported from JS long time ago and types were omited

File: src/utils/debug/texture-selector-utils.ts [hunk 5]
```diff
@@ -22,7 +22,31 @@ export function selectOriginalTextureForTileset() {
...
+  const ctx = canvas.getContext("2d");
+  if (!ctx) {
+    return null;
+  }
+
+  ctx.drawImage(bitmap, 0, 0, sw, sh);
+  return canvas;
+};
+
+export async function selectDebugTextureForTile(tile, uvDebugTexture) {
```
Comment (belom88) [inline src/utils/debug/texture-selector-utils.ts:49]: Could you add TSDoc comments and types for the function? It was ported from JS long time ago and types were omited

File: src/utils/debug/texture-selector-utils.ts [hunk 6]
```diff
@@ -42,14 +66,32 @@ export function selectDebugTextureForTile(tile, uvDebugTexture) {
       tile.userData.originalTexture =
         material.pbrMetallicRoughness.baseColorTexture.texture.source.image;
     }
+    const width =
+      material.pbrMetallicRoughness.baseColorTexture.texture.source.image.width;
+    const height =
+      material.pbrMetallicRoughness.baseColorTexture.texture.source.image
+        .height;
```
Comment (belom88) [inline src/utils/debug/texture-selector-utils.ts:73]: That should work and looks much better
```suggestion
    const width = tile.userData.originalTexture.width;
    const height = tile.userData.originalTexture.height;
```

**Review by belom88 — APPROVED** [review]

File: src/utils/debug/texture-selector-utils.ts (hunk 3 above)
Comment (mspivak-actionengine) [reply src/utils/debug/texture-selector-utils.ts]: Done

File: src/utils/debug/texture-selector-utils.ts (hunk 4 above)
Comment (mspivak-actionengine) [reply src/utils/debug/texture-selector-utils.ts]: Done

File: src/utils/debug/texture-selector-utils.ts (hunk 5 above)
Comment (mspivak-actionengine) [reply src/utils/debug/texture-selector-utils.ts]: Done

File: src/utils/debug/texture-selector-utils.ts (hunk 6 above)
Comment (mspivak-actionengine) [reply src/utils/debug/texture-selector-utils.ts]: Done

**Review by belom88 — APPROVED** [review]

**Pull request merged**